File: maxDepth.c
#define MAX(a, b) ((a) > (b) ? (a) : (b))
int maxDepth(char * s){
    int ans =0;
    int size =0;
    for(int i=0; i<strlen(s);i++){
         if (s[i] == '(') {
            ++size;
            ans = MAX(ans, size);
        } else if (s[i] == ')') {
            --size;
        }
    }
    return ans;
}

File: maxDepth.scala
object Solution {
    def maxDepth(s: String): Int = {
        var ans = 0
        var size = 0
        for ( i <-0 until s.size) //s.length{
            val ch = s.charAt(i)
            if (ch == '(') {
                size+=1
                ans = Math.max(ans, size)
            } else if (ch == ')') {
                size-=1
            }
        }
        ans
    }

}
